frontend/pages/hosts/ManageHostsPage/components/RunScriptBatchModal/helpers.ts

```diff
@@ -0,0 +1,154 @@
+import { parse, isValid } from "date-fns";
+import { IRunScriptBatchModalScheduleFormData } from "./RunScriptBatchModal";
+
+// TODO: create a validator abstraction for this and the other form validation files
+
+export interface IRunScriptBatchModalFormValidation {
+  isValid: boolean;
+  date?: { isValid: boolean; message?: string };
+  time?: { isValid: boolean; message?: string };
+}
+
+type IMessageFunc = (formData: IRunScriptBatchModalScheduleFormData) => string;
+type IValidationMessage = string | IMessageFunc;
+type IFormValidationKey = keyof Omit<
+  IRunScriptBatchModalScheduleFormData,
+  "isValid"
+>;
+
+interface IValidation {
+  name: string;
+  isValid: (
+    formData: IRunScriptBatchModalScheduleFormData,
+    validations?: IRunScriptBatchModalFormValidation
+  ) => boolean;
+  message?: IValidationMessage;
+}
+
+type IFormValidations = Record<
+  IFormValidationKey,
+  { validations: IValidation[] }
+>;
+
+const FORM_VALIDATIONS: IFormValidations = {
+  date: {
+    validations: [
+      {
+        name: "required",
+        isValid: (formData: IRunScriptBatchModalScheduleFormData) => {
+          return formData.date.length > 0;
+        },
+      },
+      {
+        name: "validDate",
+        isValid: (formData: IRunScriptBatchModalScheduleFormData) => {
+          if (!formData.date.match(/^\d{4}-\d{2}-\d{2}$/)) {
+            return false;
+          }
+          const parsedDate = parse(formData.date, "yyyy-MM-dd", new Date());
+          return isValid(parsedDate);
+        },
+        message: "Please enter a valid date.",
+      },
+      {
+        name: "notInPast",
+        isValid: (formData: IRunScriptBatchModalScheduleFormData) => {
+          const now = new Date();
+          const parsedDate = parse(
+            `${formData.date} 23:59:59.999`,
+            "yyyy-MM-dd HH:mm:ss.SSS",
+            now
+          );
+          return parsedDate >= now;
+        },
+        message: `Date cannot be in the past.`,
+      },
+    ],
+  },
+  time: {
+    validations: [
+      {
+        name: "required",
+        isValid: (formData: IRunScriptBatchModalScheduleFormData) => {
+          return formData.time.length > 0;
+        },
+      },
+      {
+        name: "validTime",
+        isValid: (formData: IRunScriptBatchModalScheduleFormData) => {
+          if (!formData.time.match(/^\d{2}:\d{2}$/)) {
+            return false;
+          }
+          const parsedDate = parse(
+            `1982-10-13 ${formData.time}`,
+            "yyyy-MM-dd HH:mm",
+            new Date()
+          );
+          return isValid(parsedDate);
+        },
+        message: "Please enter a valid time.",
```

Comment:
Looks like there's a [specific
message](https://www.figma.com/design/Q18CcrFKmgOp0f5slpH35T/-28390-Schedule-scripts-to-run-at-a-specific-time?node-id=5841-20722&m=dev)
we want here. Can probably put a similar one on the date field too?